api/envoy/extensions/common/aws/v3/credential_provider.proto

```diff
@@ -14,18 +16,26 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 
 // [#protodoc-title: AWS common configuration]
 
-// Configuration for AWS credential provider. Normally, this is optional and the credentials are
+// Configuration for AWS credential provider. This is optional and the credentials are normally
 // retrieved from the environment or AWS configuration files by following the default credential
-// provider chain. This is to support cases where the credentials need to be explicitly provided
-// by the control plane.
+// provider chain. However, this configuration can be used to override the default behavior.
 message AwsCredentialProvider {
   // The option to use `AssumeRoleWithWebIdentity <https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRoleWithWebIdentity.html>`_.
-  // If inline_credential is set, this is ignored.
-  AssumeRoleWithWebIdentityCredentialProvider assume_role_with_web_identity = 1;
+  AssumeRoleWithWebIdentityCredentialProvider assume_role_with_web_identity_provider = 1;
 
-  // The option to use an inline credential.
-  // If this is set, it takes precedence over assume_role_with_web_identity.
+  // The option to use an inline credential. If inline credential is provided, no chain will be created and only the inline credential will be used.
   InlineCredentialProvider inline_credential = 2;
+
+  // The option to specify parameters for credential retrieval from an envoy data source, such as a file in AWS credential format.
+  CredentialsFileCredentialProvider credentials_file_provider = 3;
+
+  // Create a custom credential provider chain instead of the default credential provider chain.
+  // If set to TRUE, the credential provider chain that is created contains only those set in this credential provider message.
+  // If set to FALSE, the settings provided here will act as modifiers to the default credential provider chain.
+  // Defaults to FALSE.
+  //
+  // This has no effect if inline_credential is provided.
+  bool custom_credential_provider_chain = 4;
```

Comment:
~~Is it possible to use this option to only use IMDS credentials provider and not the default chain?~~

I see code comment "Add additional providers to the custom chain". Do we want to mention today only file credentials provider is possible under the custom chain?

Reply:
I plan to add IMDS + Containers shortly after this patch goes in. At that point I will remove the additional CustomCredentialProvider factory definition